src/Optimizely/Notification/NotificationCenter.php

```diff
@@ -0,0 +1,151 @@
+    }
+
+    public function getNotifications(){
+        return $this->_notifications;
+    }
+
+    /**
+     * [addNotificationListener description]
+     * @param [type] $notification_type     [description]
+     * @param [type] $notification_callback [description]
+     */
+    public function addNotificationListener($notification_type, $notification_callback)
+    {
+        if (!NotificationType::isNotificationTypeValid($notification_type)) {
+            $this->logger->log(Logger::ERROR, "Invalid notification type.");
+            return null;
+        }
+
+        if (!is_callable($notification_callback)) {
+            $this->logger->log(Logger::ERROR, "Invalid notification callback.");
+            return null;
+        }
+
+        foreach (array_values($this->_notifications[$notification_type]) as $callback) {
+            if ($notification_callback == $callback) {
+                $this->logger->log(Logger::DEBUG, "Callback already added for notification type '{$notification_type}'.");
+                return -1;
+            }
+        }
+
+        $this->_notifications[$notification_type][$this->_notificationId] = $notification_callback;
+        $this->logger->log(Logger::INFO, "Callback added for notification type '{$notification_type}'.");
+        $returnVal = $this->_notificationId++;
+        return $returnVal;
+    }
+
+    /**
+     * [removeNotificationListener description]
+     * @param  [type] $notification_id [description]
+     * @return [type]                  [description]
+     */
+    public function removeNotificationListener($notification_id)
+    {
+        foreach ($this->_notifications as $notification_type => $notifications) {
+            foreach (array_keys($notifications) as $id) {
+                if ($notification_id == $id) {
+                    unset($this->_notifications[$notification_type][$id]);
+                    $this->logger->log(Logger::INFO, "Callback with notification ID '{$notification_id}' has been removed.");
+                    return true;
+                }
+            }
+        }
+
+        $this->logger->log(Logger::DEBUG, "No Callback found with notification ID '{$notification_id}'.");
+        return false;
+    }
+
+    /**
+     * [clearNotifications description]
+     * @param  [type] $notification_type [description]
+     * @return [type]                    [description]
+     */
+    public function clearNotifications($notification_type)
+    {
+        if (!NotificationType::isNotificationTypeValid($notification_type)) {
+            $this->logger->log(Logger::ERROR, "Invalid notification type.");
+            return null;
+        }
+
+        $this->_notifications[$notification_type] = [];
+        $this->logger->log(Logger::INFO, "All callbacks for notification type '{$notification_type}' have been removed.");
+    }
+
+    /**
+     * [clearAllNotifications description]
+     * @return [type] [description]
+     */
+    public function clearAllNotifications()
```

Comment:
sounds weird, but it should be clean :D

Reply:
Please confirm the naming convention. 
The other method is clearNotification($type). So this should be clearAllNotifications

Reply:
in python it is clean